storm-core/src/jvm/backtype/storm/metric/internal/LatencyStatAndMetric.java

```diff
@@ -242,7 +248,7 @@ private synchronized void rotate(long lat, long count, long timeSpent, long targ
             totalCount += countBuckets[i];
             timeNeeded -= bucketTime[i];
         }
-        return ((double)totalLat)/totalCount;
+        return totalCount > 0 ? ((double)totalLat)/totalCount : 0.0;
```

Comment:
Maybe we can pull the code out into a function named like `div-or-0`, as it appears more than once.